File: BingoManager.Reports/Viewer/ReportViewer.vb
Imports BingoManager.SystemManager.Engine
Imports System.Data.OleDb
Imports BingoManager.Reports.bingoDataSetTableAdapters


Public Class ReportViewer
    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.

    End Sub


    Private Sub ReportViewer_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Using connection = ConnectionProvider.Connection()
            Dim sqlCommand As String = "Select SerialNumber,B,I,N,G,O from PlayingCards"
            Dim da As New OleDbDataAdapter(sqlCommand, connection)
             Dim bingodataset As New bingoDataSet()
            da.Fill(bingodataset, "PlayingCards")
            Dim rePort As New CrystalReport1()
            rePort.SetDataSource(bingodataset)
            CrystalReportViewer1.ReportSource = rePort
            CrystalReportViewer1.Refresh()
        End Using
    End Sub
End Class

File: BingoManager.Report/View/ReportViewer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Data.OleDb;

using BingoManager.SystemManager.Engine;
using BingoManager.SystemManager.Model;
using BingoManager.SystemManager.Reports;

namespace BingoManager.SystemManager.View
{
    public partial class ReportViewer : Form
    {
        public ReportViewer()
        {
            InitializeComponent();
            this.Load += new EventHandler(ReportViewer_Load);
        }

        void ReportViewer_Load(object sender, EventArgs e)
        {
            try
            {
                using (OleDbConnection connection = ConnectionProvider.Connection())
                {
                    string sqlCommand = "Select SerialNumber,B,I,N,G,O From PlayingCards;";
                    using (OleDbDataAdapter da = new OleDbDataAdapter(sqlCommand, connection))
                    {
                        bingoDataSet ds = new bingoDataSet();
                       da.Fill(ds, "PlayingCards");
                        CrystalReport1 report = new CrystalReport1();
                        report.SetDataSource(ds);
                        crystalReportViewer1.ReportSource = report;
                        crystalReportViewer1.Refresh();
                    }
                }
            }
            catch (Exception) { }
        }

    }
}
